File: src/org/cappuccino_project/ide/intellij/plugin/lang/ObjJFileTypeFactory.java
package org.cappuccino_project.ide.intellij.plugin.lang;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.cappuccino_project.ide.intellij.plugin.extensions.plist.ObjJPlistFile;
import org.cappuccino_project.ide.intellij.plugin.extensions.plist.ObjJPlistFileType;
import org.jetbrains.annotations.NotNull;

public class ObjJFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull
                                        FileTypeConsumer fileTypeConsumer) {
        fileTypeConsumer.consume(ObjJFileType.INSTANCE, ObjJFileType.FILE_EXTENSION);
        fileTypeConsumer.consume(ObjJPlistFileType.INSTANCE, ObjJPlistFileType.FILE_EXTENSION);
    }
}

File: src/cappuccino/ide/intellij/plugin/lang/ObjJFileTypeFactory.kt
@file:Suppress("DEPRECATION")

package cappuccino.ide.intellij.plugin.lang

import cappuccino.ide.intellij.plugin.jstypedef.lang.JsTypeDefFileType
import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class ObjJFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        fileTypeConsumer.consume(ObjJFileType.INSTANCE, ObjJFileType.FILE_EXTENSION)
        fileTypeConsumer.consume(JsTypeDefFileType.INSTANCE, JsTypeDefFileType.FILE_EXTENSION)
        //fileTypeConsumer.consume(ObjJPlistFileType.instance, ObjJPlistFileType.FILE_EXTENSION);
    }
}
